src/main/java/org/tikv/txn/TikvTransaction.java

```diff
@@ -0,0 +1,221 @@
+package org.tikv.txn;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tikv.common.Snapshot;
+import org.tikv.common.key.Key;
+import org.tikv.common.meta.TiTimestamp;
+
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Transaction implementation of TiKV client
+ */
+public class TikvTransaction implements ITransaction {
```

Comment:
Do we need `scan()`? If scan is needed, I think we will have to provide `scan` in `TikvTransaction` to merge keys in `memoryKv` and remote keys.

Reply:
... And I believe we should reach an agreement on `TiKV`/`Tikv`.